openvdb_houdini/SOP_OpenVDB_Extrapolate.cc

```diff
@@ -230,29 +259,44 @@ SOP_OpenVDB_Extrapolate::Cache::process(
     hvdb::GU_PrimVDB* lsPrim,
     fpreal time)
 {
-    typename GridT::Ptr outGrid; 
     typename GridT::ConstPtr inGrid = openvdb::gridConstPtrCast<GridT>(lsPrim->getConstGridPtr());
+    typename GridT::Ptr outGrid;
 
     UT_String mode;
     evalString(mode, "mode", 0, time);
     const int nSweeps = static_cast<int>(evalInt("sweeps", 0, time));
+    const float isoValue = evalFloat("isovalue", 0, time);
 
     using namespace openvdb::tools;
     if (mode == "mask") {
-        FastSweepMaskOp<GridT> op(inGrid, evalInt("tiles", 0, time), nSweeps);
+        FastSweepMaskOp<GridT> op(inGrid, evalInt("ignoretiles", 0, time), nSweeps);
         UTvdbProcessTypedGridTopology(UTvdbGetGridType(*maskGrid), *maskGrid, op);
         outGrid = op.outGrid;
     } else if (mode == "dilate") {
         UT_String str;
-        evalString(str, "nn", 0, time);
+        evalString(str, "pattern", 0, time);
         const NearestNeighbors nn =
             (str == "NN18") ? NN_FACE_EDGE : ((str == "NN26") ? NN_FACE_EDGE_VERTEX : NN_FACE);
         outGrid = dilateSdf(*inGrid, static_cast<int>(evalInt("dilate", 0, time)), nn, nSweeps);
     } else if (mode == "convert") {
-        // TODO: Confirm that convert is meant to be used for fogToSdf. Previously, the API call
-        // is maskSDF(*outGrid, evalFloat("isovalue", 0, time), evalInt("tiles", 0, time), nSweeps);
-        outGrid = fogToSdf(*inGrid, evalFloat("isovalue", 0, time), nSweeps);
+        outGrid = fogToSdf(*inGrid, isoValue, nSweeps);
         lsPrim->setVisualization(GEO_VOLUMEVIS_ISO, lsPrim->getVisIso(), lsPrim->getVisDensity());
+    } else if (mode == "correct") {
+        outGrid = sdfToSdf(*inGrid, isoValue, nSweeps);
+    } else if (mode == "fogext") {
+        // TODO: op
+        // outGrid = fogToExt(*inGrid, op, isoValue, nSweeps);
```

Comment:
@jmlait: Jeff, do you have a suggestion on how to expose the functor to the users in Houdini? According to the documentation, the functor defines the Dirichlet boundary condition and its signature is [](const Vec3R &xyz)->float.

Reply:
I think the most common use case is the user provides a VDB to provide the values of the functor.  Having a parameter callback for the functor would likely be too slow, and a VDB lets you pre-populate the isosurface with your favorite functor in an efficient manner.